packages/cli/src/services/last-active-at.service.ts

```diff
@@ -0,0 +1,27 @@
+import { UserRepository } from '@n8n/db';
+import { Service } from '@n8n/di';
+
+import { Time } from '@/constants';
+
+const LAST_ACTIVE_CACHE_TTL = 2 * Time.minutes.toMilliseconds;
+
+@Service()
+export class LastActiveAtService {
+	private readonly lastActiveCache = new Map<string, number>();
+
+	constructor(private readonly userRepository: UserRepository) {}
+
+	async updateLastActiveIfStale(userId: string) {
```

Comment:
Are you sure that this will not incur quite some load onto the database? I think we said, that it would be good enough to update date it on a daily basis. I suggested this, because I was afraid that for large installations, we would produce a lot of writes to the database.

Reply:
ah yes, so you would say change the LAST_ACTIVE_CACHE_TTL to one day (or maybe 1 hour is enough)

Reply:
@afitzek I changed the TTL to 1 hour for now